src/utils/check-pr/check-pr.py

```diff
@@ -0,0 +1,319 @@
+        self.pr = pr
+        self.branch = branch
+        self.instance = instance
+        self.region = region
+        self.subscription_id = subscription_id
+        self.skip_tests = skip_tests
+        self.test_args = test_args
+
+    def merge(self) -> None:
+        if self.pr:
+            self.downloader.merge_pr(self.branch, self.pr)
+
+    def deploy(self, filename: str) -> None:
+        print(f"deploying {filename} to {self.instance}")
+        venv = "deploy-venv"
+        subprocess.check_call(f"python -mvenv {venv}", shell=True)
+        pip = venv_path(venv, "pip")
+        py = venv_path(venv, "python")
+        commands = [
+            ("extracting release-artifacts", f"unzip -qq {filename}"),
+            ("extracting deployment", "unzip -qq onefuzz-deployment*.zip"),
+            ("installing wheel", f"{pip} install -q wheel"),
+            ("installing prereqs", f"{pip} install -q -r requirements.txt"),
+            (
+                "running deployment",
+                (
+                    f"{py} deploy.py {self.region} "
+                    f"{self.instance} {self.instance} cicd"
+                    f" {' --subscription_id ' + self.subscription_id if self.subscription_id else ''}"
+                ),
+            ),
+        ]
+        for (msg, cmd) in commands:
+            print(msg)
+            subprocess.check_call(cmd, shell=True)
+
+    def test(self, filename: str) -> None:
+        venv = "test-venv"
+        subprocess.check_call(f"python -mvenv {venv}", shell=True)
+        py = venv_path(venv, "python")
+        test_dir = "integration-test-artifacts"
+        script = "integration-test.py"
+        endpoint = f"https://{self.instance}.azurewebsites.net"
+        commands = [
+            (
+                "extracting integration-test-artifacts",
+                f"unzip -qq {filename} -d {test_dir}",
+            ),
+            ("test venv", f"python -mvenv {venv}"),
+            ("installing wheel", f"./{venv}/bin/pip install -q wheel"),
+            ("installing sdk", f"./{venv}/bin/pip install -q sdk/*.whl"),
+            (
+                "running integration",
+                (
+                    f"{py} {test_dir}/{script} test {test_dir} "
+                    f"--region {self.region} --endpoint {endpoint} "
+                    " ".join(self.test_args)
+                ),
+            ),
+        ]
+        for (msg, cmd) in commands:
+            print(msg)
+            subprocess.check_call(cmd, shell=True)
+
+    def cleanup(self, skip: bool) -> None:
+        os.chdir(tempfile.gettempdir())
+        if skip:
+            return
+
+        cmd = ["az", "group", "delete", "-n", self.instance, "--yes", "--no-wait"]
+        print(cmd)
+        subprocess.call(cmd)
+        print("done")
+
+    def run(self, *, merge_on_success: bool = False) -> None:
+        release_filename = "release-artifacts.zip"
+        self.downloader.get_artifact(
+            "microsoft/onefuzz",
```

Comment:
This is getting the artifact from the onefuzz repo right? can we make it parameter so that we can use on other repo

Reply:
added